src/Sylius/Bundle/ApiBundle/SectionResolver/AdminApiUriBasedSectionResolver.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\SectionResolver;
+
+use Sylius\Bundle\CoreBundle\SectionResolver\SectionCannotBeResolvedException;
+use Sylius\Bundle\CoreBundle\SectionResolver\SectionInterface;
+use Sylius\Bundle\CoreBundle\SectionResolver\UriBasedSectionResolverInterface;
+
+final class AdminApiUriBasedSectionResolver implements UriBasedSectionResolverInterface
```

Comment:
Just a rough idea - these `UriBasedSectionsResolver`s have generally the same implementation, it only differs regarding the returned class (still, the interface is the same). Maybe it would be worth extracting some generic resolver from them? 🤔  Not a block for merge, just thinking out loud 🖖